### DISCUSSION TITLE
Should Router firmware use port /dev/ttyACM0? I seem to have /dev/ttyAMA0 [not working]

### DISCUSSION THREAD

**Discussion by alan-rpi** [Koenkk/zigbee2mqtt#7998] [Devices]
Hi. Installing on a new Raspi Zero with Rasp Pi OS Lite 32bit version 10+deb10u2+rpt1 (Buster) and latest Zigbee2MQTT version 1.20.0 (commit #41b67fdd) and zigbee-herdsman (0.13.117)

I have a CC2531 accessed through a combined USB Hub/Ethernet Port. I want this CC2531 to be a Router.

Initially I incorrectly flashed the CC2531 with CC2531_DEFAULT_20201127 (CC2531ZNP-Prod.hex) thinking there was now a combined Coordinator/Router firmware. This did run OK with port: /dev/ttyACM0 though it found no devices (there is another CC2530 running as Coordinator).

I then found and flashed CC2531_router_2020_09_29 (router-cc2531-std.hex) and pressed the reset button S1 to get a green LED. But now I get "Error while opening serialport"

ls -l /dev/serial/by-id    gives   No such file or directory
ls -l /dev/ttyA*             gives    crw--w---- 1 root tty 204, 64 Jul  9 18:46 /dev/ttyAMA0

There is also a /dev/serial0 -> ttyAMA0  so I changed the to port: /dev/ttyAMA0 but now I get "Error: Permission denied, cannot open /dev/ttyAMA0"

Before trying to resolve this; in case AMA0 is another device, I would like clarification whether the Router firmware should also be found as /dev/ttyACM0 . If so, why isn't it in my case? And if AMA0 is correct, then why am I getting the Permission denied for the Router and not the Coordinator?

Help appreciated. 
Alan

**Comment by Koenkk** [comment]
The router firmware does not have an USB interface, it only needs power to work (so just connecting it and it will work

**Comment by alan-rpi** [comment]
Thanks for replying Koen. 

So the Router firmware cannot work with zigbee2MQTT on the same device - is that correct? 

My CC2531 is on a Pi Zero with Mosquitto and zigbee2MQTT installed and in effect acting as an End Device - I already have a Coordinator running on a another Pi4 with Mosquitto and zigbee2MQTT (the Pi0 and Pi4 and not connected via Ethernet).

Maybe the Pi0 with Xbee is the answer, or CC2530 with a PVTO and a UART connection. 

It would be nice if you could make End Device firmware that communicated with zigbee2MQTT (i.e. Coordinator firmware with the coordination function switched off)
